models/meshmodel/core/v1alpha1/relationship.go

```diff
@@ -73,7 +73,7 @@ func GetMeshModelRelationship(db *database.Handler, f RelationshipFilter) (r []R
 		if f.Greedy {
 			finder = finder.Where("relationship_definition_dbs.kind LIKE ?", "%"+f.Kind+"%")
 		} else {
-			finder = finder.Where("relationship_definition_dbs.kind = ?", "%"+f.Kind+"%")
```

Comment:
@abdullah1308 the need for this non-greedy change is not clear to me.

Reply:
@leecalcote Line 74 is for when we want the search to be greedy. Line 76 is for when we want the search to not be greedy. 

When using `=`, using `"%"+f.Kind+"%"` doesn't work since it searches for the string literal `%Kind%` which wrong as we want it to search for `Kind`.

Reply:
https://github.com/meshery/meshkit/blob/403bdf4f6bed995324c8348b1e4954163f798152/models/meshmodel/core/v1alpha1/component.go#L95C1-L110